flocker/node/test/test_change.py

```diff
@@ -521,3 +523,47 @@ def test_failed(self, logger):
         action._logger = logger
         failure = self.failureResultOf(run_state_change(action, DEPLOYER))
         self.assertEqual(failure.getErrorMessage(), "Oh no")
+
+    def assert_nested_logging(self, combo, action_type, logger):
+        """
+        All the underlying ``IStateChange`` will be run in Eliot context in
+        which the sequential ``IStateChange`` is run, even if they are not
+        run immediately.
+
+        :param combo: ``sequentially`` or ``in_parallel``.
+        :param action_type: ``eliot.ActionType`` we expect to be parent of
+            sub-changes logs.
+        :param logger: A ``MemoryLogger`` where messages go.
+        """
+        actions = [ControllableAction(result=Deferred()),
+                   ControllableAction(result=succeed(None))]
+        for action in actions:
+            self.patch(action, "_logger", logger)
+        run_state_change(combo(actions), None)
+        # For sequentially this will ensure second action doesn't
+        # automatically run in context of LOG_ACTION:
+        actions[0].result.callback(None)
+
+        parent = assertHasAction(self, logger, action_type, succeeded=True)
+        self.assertEqual(
+            dict(messages=parent.children, length=len(parent.children)),
+            dict(
+                messages=LoggedAction.ofType(
+                    logger.messages, CONTROLLABLE_ACTION_TYPE),
+                length=2))
+
+    @capture_logging(None)
+    def test_sequential_logging(self, logger):
+        """
+        All the underlying ``IStateChange`` will be run in Eliot context in
+        which the sequential ``IStateChange`` is run.
+        """
+        self.assert_nested_logging(sequentially, LOG_SEQUENTIALLY, logger)
```

Comment:
:grey_question: No action required, but wanted to get this off my chest. 

I'm not a fan of the pattern of making one big helper that does the set-up, operation to test and assertions. What is saved in repetition, is often lost in ease of manual analysis and  cyclomatic simplicity. In cases like line 545, code that's not relevant to one test is run as a part of another, which (in other circumstances) invites unexpected dependencies of behaviour. 

I'm all for helpers for set-up and for domain-specific assertion methods, but I think these should be called explicitly from within `test_` methods, and that manipulation of the system under test should likewise be direct from test methods.

I realize though that "one big helper" is a very common pattern in flocker, and that this review isn't the place to change that. So, no change required.